core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -136,15 +181,92 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   def reset(): Unit
 
   /**
+   * Takes the inputs and accumulates. e.g. it can be a simple `+=` for counter accumulator.
+   * Developers should extend addImpl to customize the adding functionality.
+   */
+  final def add(v: IN): Unit = {
+    if (metadata != null && metadata.dataProperty) {
+      dataPropertyAdd(v)
+    } else {
+      addImpl(v)
+    }
+  }
+
+  private def dataPropertyAdd(v: IN): Unit = {
+    // To allow the user to be able to access the current accumulated value from their process
+    // worker side then we need to perform a "normal" add as well as the data property add.
+    addImpl(v)
+    // Add to the pending updates for data property
+    val updateInfo = TaskContext.get().getRDDPartitionInfo()
+    val base = pending.getOrElse(updateInfo, copyAndReset())
```

Comment:
rename `base` to `mergedAccumulatorForOneTaskOutput` (or something like that?  I just find `base` way to generic.)